src/battle_util.c

```diff
@@ -4060,6 +4060,58 @@ static inline bool32 HadMoreThanHalfHpNowDoesnt(u32 battler)
              && gBattleMons[battler].hp <= cutoff);
 }
 
+#define ANIM_STAT_STAT_HP      0
+#define ANIM_STAT_STAT_ATK     1
+#define ANIM_STAT_STAT_DEF     2
+#define ANIM_STAT_STAT_SPATK   3
+#define ANIM_STAT_STAT_SPDEF   4
+#define ANIM_STAT_STAT_SPEED   5
+#define ANIM_STAT_STAT_ACC     6
+#define ANIM_STAT_STAT_EVASION 7
+static void ChooseStatBoostAnimation(u32 battler)
+{
+    u32 stat;
+    bool32 statBuffMoreThan1 = FALSE;
+    u32 static const statsOrder[NUM_BATTLE_STATS] =
+    {
+        [ANIM_STAT_STAT_HP]      = STAT_HP,
+        [ANIM_STAT_STAT_ATK]     = STAT_ATK,
+        [ANIM_STAT_STAT_DEF]     = STAT_DEF,
+        [ANIM_STAT_STAT_SPATK]   = STAT_SPATK,
+        [ANIM_STAT_STAT_SPDEF]   = STAT_SPDEF,
+        [ANIM_STAT_STAT_SPEED]   = STAT_SPEED,
+        [ANIM_STAT_STAT_ACC]     = STAT_ACC,
+        [ANIM_STAT_STAT_EVASION] = STAT_EVASION,
+    };
+    gBattleScripting.animArg1 = 0;    
+
+    for (stat = 0; stat < NUM_BATTLE_STATS; stat++)
+    {
+        if (stat == STAT_HP) // Skip HP because it can not be boosted
+            continue;
```

Comment:
Isn't this a wasted iteration considering HP is always 0 anyway? Shouldn't we just have `stat` start at `1` or even explicitly `STAT_ATK`?

Reply:
right, that is better